server/services/auth/password_test.go

```diff
@@ -135,7 +135,10 @@ func TestIsPasswordValidWithSettings(t *testing.T) {
 				assert.NoError(t, err)
 			} else {
 				require.Error(t, err)
-				assert.Equal(t, tc.ExpectedFailingCriterias, err.(*InvalidPasswordError).FailingCriterias)
+				var errFC *InvalidPasswordError
+				if assert.ErrorAs(t, err, &errFC) {
```

Comment:
nit: should we assert that the error type is `InvalidPasswordError`. If `IsPasswordValid` start returning something that is not `InvalidPasswordError`, it would pass the tests, but not necesarilly means that is right.

Reply:
`assert.ErrorAs` does check that err is of type `InvalidPasswordError`.  It will fail if err does not have that type somewhere in the chain of wrapped errors.